### PR TITLE
feat: Added django3.2 support for production deployment.

### PR DESCRIPTION
**Pull request by awais786**
https://openedx.atlassian.net/browse/BOM-2653

- Added django3.2 for production.
- Dropped django22 and others from testing.
- Fixed few warnings.
- make upgrade

CheckList.
- [x] Tested locally
- [x] Run migrate command
- [x] Run makemigrations command ( no new migration )

Site seems stable locally.

Screen Shot 2021-07-25 at 4 37 20 PM

### CODE REVIEW

File: openedxstats/settings/base.py [hunk 1]
```diff
@@ -97,6 +97,7 @@
 
 USE_TZ = False
 
+DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
```
Comment (awais786) [inline openedxstats/settings/base.py:100]:
```
HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
sites.FilenameLog: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
```

File: openedxstats/settings/base.py [hunk 2]
```diff
@@ -128,6 +129,7 @@
             'context_processors': [
                 'django.contrib.auth.context_processors.auth',
                 'django.template.context_processors.debug',
+                'django.template.context_processors.request',
```
Comment (awais786) [inline openedxstats/settings/base.py:132]:
```?: (admin.W411) 'django.template.context_processors.request' must be 
enabled in DjangoTemplates (TEMPLATES) in order to use the admin navigation 
sidebar.
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -20,9 +20,6 @@ jobs:
         python-version:
           - 3.8
         toxenv:
-          - django22
-          - django30
-          - django31
```
Comment (jmbowman) [inline .github/workflows/ci.yml:25]: Dropping testing for these concurrently with the 3.2 upgrade deployment is probably fine for an infrequently changing repo like this one, but I'd make it a separate step on something like edx-platform where it's entirely possible that a new PR could introduce incompatibilities with 2.2 between the time the upgrade PR merges and determining if a rollback from stage or prod is needed.

File: tox.ini [hunk 4]
```diff
@@ -1,12 +1,9 @@
...
-envlist = py38-django{22,30,31,32}
+envlist = py38-django{32}
 skipsdist = true
 
 [testenv]
 deps =
-    django22: -r{toxinidir}/requirements/django.txt
-    django30: Django>=3.0,<3.1
-    django31: Django>=3.1,<3.2
     django32: Django>=3.2,<3.3
```
Comment (jmbowman) [inline tox.ini:7]: This one should be updated to use `-r{toxinidir}/requirements/django.txt`.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (awais786) [reply .github/workflows/ci.yml:25]: I will add these steps back.

**Review by mzulqarnain1 — APPROVED** [review]

File: requirements/constraints.txt [hunk 5]
```diff
@@ -8,8 +8,6 @@
 # pin when possible.  Writing an issue against the offending project and
 # linking to it here is good.
 
-# Common constraints for edx repos
--c https://raw.githubusercontent.com/edx/edx-lint/master/edx_lint/files/common_constraints.txt
- 
+
```
[the remark this answers is not in the document]
Comment (awais786) [reply requirements/constraints.txt]: Using this syncing common constraint locally approach. we have a backlog ticket to finalize this. For time being we can use this now.

**Comment by awais786** [comment]
@nedbat Please review.

**Comment by awais786** [comment]
> @awais786 can you rebase?

@nedbat I have rebased.

**Comment by awais786** [comment]
@nedbat Please review.

**Review by nedbat — APPROVED** [review]

**Comment by nedbat** [comment]
I'm merging it so I can deploy it while it's top-of-mind.

**Pull request merged**